pandas/core/indexes/base.py

```diff
@@ -499,10 +501,12 @@ def _shallow_copy(self, values=None, name: Label = no_default):
         """
         name = self.name if name is no_default else name
 
-        if values is None:
-            values = self.values
+        cache = self._cache if values is None else {}
```

Comment:
can you keep the `if values is None:` check; i like it for inspection of coverage results